hs_core/models.py

```diff
@@ -1354,6 +1303,77 @@ class AbstractResource(ResourcePermissionsMixin):
         class MyResourceContentType(pages.Page, hs_core.AbstractResource):
             ...
     """
+    # For each request for a resource page view, the Page will run can_view(), can_change(),
+    # can_delete() in succession to check for permissions, so for performance improvement,
+    # there might be merit to make ha_obj a class member variable with a persistent connection
+    # for the request, rather than a separate connection within each method call
+    def can_add(self, request):
+        return self.can_change(request)
+
+    def can_delete(self, request):
+        user = get_user(request)
+        if user.is_authenticated():
+            if user.is_superuser:
+                ret = True
+            elif self.creator and user.pk == self.creator.pk:
+                ret = True
+            else:
+                try:
+                    ha_obj = HSAlib.HSAccess(str(user.username), 'unused', settings.HS_ACCESS_DB, settings.HS_ACCESS_USERNAME,
+                                             settings.HS_ACCESS_PASSWORD, settings.HS_ACCESS_HOST, settings.HS_ACCESS_PORT)
+                    ret = ha_obj.resource_is_owned(self.short_id)
+                    del ha_obj
+                except:
```

Comment:
Hmm...The logic behind this "try..except" is: if there is any issue raised when checking whether the resource is owned, just return false, although this except should not happen unless something unexpected happened and things really went awry in some other places. On second thought, I agree it'd be better to raise an exception here before returning false, but in some other circumstances, for example, if an underlying library like gdal cannot extract transform info due to invalid input and raises an exception, and we want to catch that exception but not show the exception message, but rather just set the metadata to default NA values since the valid metadata cannot be extracted. In any case, I agree bare except should be used cautiously, but is needed in certain circumstances in my opinion. Will add code to raise exceptions in this specific case. Thanks!